lib/Parse/Lexer.cpp

```diff
@@ -1843,9 +1843,30 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
   assert((QuoteChar == '"' || QuoteChar == '\'') && "Unexpected start");
 
   bool IsMultilineString = advanceIfMultilineDelimiter(CurPtr, Diags);
-  if (IsMultilineString && *CurPtr != '\n' && *CurPtr != '\r')
-    diagnose(CurPtr, diag::lex_illegal_multiline_string_start)
-        .fixItInsert(Lexer::getSourceLoc(CurPtr), "\n");
+  if (IsMultilineString && *CurPtr != '\n' && *CurPtr != '\r') {
+    // Test for single-line string literals that may resemble multiline delimiter.
+    for (const char *Ptr = CurPtr; Ptr <= BufferEnd-CustomDelimiterLen; Ptr++) {
+      if (*Ptr == '\r' || *Ptr == '\n') {
+        break;
+      }
+      if (*Ptr == '#') {
```

Comment:
Please check the existence of `"` before this. For example: `#"""     This is #1.` In this case, we want to diagnose it as `multi-line string literal content must begin on a new line` instead of single line `unterminated string literal`.